elmslie-android/src/main/java/vivid/money/elmslie/android/fabric/StoreFabrics.kt

```diff
@@ -0,0 +1,105 @@
+package vivid.money.elmslie.android.fabric
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.annotation.MainThread
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.AbstractSavedStateViewModelFactory
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.savedstate.SavedStateRegistryOwner
+import vivid.money.elmslie.core.store.Store
+
+@MainThread
+fun <
+    Event : Any,
+    Effect : Any,
+    State : Any,
+> elmStore(
+    key: String,
+    getViewModelStoreOwner: () -> ViewModelStoreOwner,
+    getSavedStateRegistryOwner: () -> SavedStateRegistryOwner,
+    getDefaultArgs: () -> Bundle,
+    storeFactory: (SavedStateHandle) -> Store<Event, Effect, State>,
+): Lazy<Store<Event, Effect, State>> =
+    lazy(LazyThreadSafetyMode.NONE) {
+        val factory =
+            RetainedElmStoreFactory(
+                stateRegistryOwner = getSavedStateRegistryOwner.invoke(),
+                defaultArgs = getDefaultArgs.invoke(),
+                storeFactory = storeFactory,
+            )
+        val provider = ViewModelProvider(getViewModelStoreOwner.invoke(), factory)
+
+        @Suppress("UNCHECKED_CAST")
+        provider.get(key, RetainedElmStore::class.java).store as Store<Event, Effect, State>
+    }
+
+@MainThread
+fun <
+    Event : Any,
+    Effect : Any,
+    State : Any,
+> Fragment.elmStore(
+    key: String = this::class.java.canonicalName ?: this::class.java.simpleName,
+    getViewModelStoreOwner: () -> ViewModelStoreOwner = { this },
+    getSavedStateRegistryOwner: () -> SavedStateRegistryOwner = { this },
+    getDefaultArgs: () -> Bundle = { arguments ?: Bundle() },
+    storeFactory: (SavedStateHandle) -> Store<Event, Effect, State>,
+): Lazy<Store<Event, Effect, State>> =
+    vivid.money.elmslie.android.fabric.elmStore(
```

Comment:
can be short?

Reply:
nope, short option will be reference to the same method